packages/shared/src/features/onboarding/shared/DiscountTimer.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import type { ReactElement } from 'react';
+import classNames from 'classnames';
+import createDOMPurify from 'dompurify';
+
+/**
+ * Formats seconds to MM:SS format
+ */
+const formatTime = (seconds: number): string => {
+  if (seconds <= 0) {
+    return '00:00';
+  }
+
+  const minutes = Math.floor(seconds / 60);
+  const remainingSeconds = seconds % 60;
+
+  return `${minutes.toString().padStart(2, '0')}:${remainingSeconds
+    .toString()
+    .padStart(2, '0')}`;
+};
```

Comment:
Maybe we could use `date-fns`, don't know if there is a function ready for it

Reply:
There's none from what I checked now